File: hugin/example/imshow.ml
open Hugin

let () =
  if Array.length Sys.argv < 2 then (
    Printf.printf "Usage: %s <image_path>\n" Sys.executable_name;
    exit 1);

  let image_path = Sys.argv.(1) in
  let img = Ndarray_io.load_image image_path in
  let fig = imshow ~title:"Image" img in
  show fig

File: hugin/examples/02-imshow/imshow.py
import sys
import cv2
import numpy as np
import matplotlib.pyplot as plt

if len(sys.argv) < 2:
    print(f"Usage: {sys.argv[0]} <image_path>")
    sys.exit(1)

image_path = sys.argv[1]
img = cv2.imread(image_path)
# Convert BGR (OpenCV default) to RGB for proper display in matplotlib
img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)

plt.figure()
plt.imshow(img)
# plt.title("Image")
# plt.savefig("output.png")
# plt.close()
plt.show()